rdkit/Chem/Subshape/SubshapeAligner.py

```diff
@@ -282,42 +274,44 @@ def PruneMatchesUsingShape(self, targetMol, target, queryMol, query, builder, al
   def GetSubshapeAlignments(self, targetMol, target, queryMol, query, builder, tgtConf=-1,
                             queryConf=-1, pruneStats=None):
     import time
+
     if pruneStats is None:
       pruneStats = {}
+
     logger.info("Generating triangle matches")
-    t1 = time.time()
+    t1 = time.perf_counter()
     res = [x for x in self.GetTriangleMatches(target, query)]
-    t2 = time.time()
-    logger.info("Got %d possible alignments in %.1f seconds" % (len(res), t2 - t1))
-    pruneStats['gtm_time'] = t2 - t1
+    pruneStats['gtm_time'] = time.perf_counter() - t1
+    logger.info("Got %d possible alignments in %.1f seconds" % (len(res), pruneStats['gtm_time']))
+
     if builder.featFactory:
       logger.info("Doing feature pruning")
-      t1 = time.time()
+      t1 = time.perf_counter()
       self.PruneMatchesUsingFeatures(target, query, res, pruneStats=pruneStats)
-      t2 = time.time()
-      pruneStats['feats_time'] = t2 - t1
-      logger.info("%d possible alignments remain. (%.1f seconds required)" % (len(res), t2 - t1))
+      pruneStats['feats_time'] = time.perf_counter() - t1
+      logger.info("%d possible alignments remain. (%.1f seconds required)" % (len(res), pruneStats['feats_time']))
     logger.info("Doing direction pruning")
-    t1 = time.time()
+
+    t1 = time.perf_counter()
     self.PruneMatchesUsingDirection(target, query, res, pruneStats=pruneStats)
-    t2 = time.time()
-    pruneStats['direction_time'] = t2 - t1
-    logger.info("%d possible alignments remain. (%.1f seconds required)" % (len(res), t2 - t1))
-    t1 = time.time()
+    pruneStats['direction_time'] = time.perf_counter() - t1
+    logger.info("%d possible alignments remain. (%.1f seconds required)" % (len(res), pruneStats['direction_time']))
+
+    t1 = time.perf_counter()
     self.PruneMatchesUsingShape(targetMol, target, queryMol, query, builder, res, tgtConf=tgtConf,
                                 queryConf=queryConf, pruneStats=pruneStats)
-    t2 = time.time()
-    pruneStats['shape_time'] = t2 - t1
+    pruneStats['shape_time'] = time.perf_counter() - t1
     return res
 
   def __call__(self, targetMol, target, queryMol, query, builder, tgtConf=-1, queryConf=-1,
                pruneStats=None):
     for alignment in self.GetTriangleMatches(target, query):
-      if (not self._checkMatchFeatures(target.skelPts, query.skelPts, alignment) and
-          builder.featFactory):
-        if pruneStats is not None:
-          pruneStats['features'] = pruneStats.get('features', 0) + 1
-        continue
+      if builder.featFactory:
```

Comment:
This code doesn't do the same thing as the original.

The original code always called _checkMatchFeatures before checking if featFactory existed.  This checks featFactory first.

Reply:
I know that the old method do `_checkMatchFeatures()` first at the time refactoring. However, as said, if the function `_checkMatchFeatures()` doing ahead of the design, it would be the damaging of function design rather than the naive optimization. I have discussed this type in PR #4815. 
As discussed, this type of PR do naive-blindful optimization, which means that given unknown input and processing (little datatype is known and guessed), how much optimization we can do ahead.

In the end, all unit tests passed and zero issues raised.
@gedeck What do you think?

Reply:
While the test coverage is reasonable on the Python side, I wouldn't assume that a piece of code is covered by tests. This means passing tests doesn't necessarily imply correctness of refactoring.

Reply:
> This code doesn't do the same thing as the original.
> 
> The original code always called _checkMatchFeatures before checking if featFactory existed. This checks featFactory first.

If you worrried about this change, I will revert back to the original

Reply:
OK, I got your idea! I would fix it in the next patch. Thank you for your review.